### PR TITLE
CSCQVAIN-318: Do not redirect user after publish.

### PR DESCRIPTION
**Pull request by jppiiroinen**
Show the Etsin button instead. This will cause less confusion for the users.

### CODE REVIEW

File: src/views/Editor.vue [hunk 1]
```diff
@@ -141,6 +153,16 @@
 					<router-view></router-view>
 				</b-col>
 
+				<b-col v-else-if="loading">
+					<b-container>
+						<b-row>
+							<b-col>
+								<font-awesome-icon icon="spinner" spin />
```
Comment (jppiiroinen) [inline src/views/Editor.vue:159]: This fixes the issue with missing loading indicator on Editor page load.

File: src/views/Editor.vue [hunk 2]
```diff
@@ -260,6 +282,9 @@ export default {
 			}
 			return null
 		},
+		viewInEtsin() {
+			window.open(`${process.env.VUE_APP_ETSIN_API_URL}/${this.qvainData.identifier}`, '_blank')
```
Comment (jppiiroinen) [inline src/views/Editor.vue:302]: This requires the API change which is in the other PR.

File: src/views/Editor.vue [hunk 3]
```diff
@@ -498,7 +531,7 @@ export default {
 	async mounted() {
 		if (this.id === 'new') {
 			this.clearRecord()
-		} else if (this.id !== 'edit' && this.$store.state.metadata.id !== this.id) {
+		} else if (this.id !== 'edit') {
```
Comment (jppiiroinen) [inline src/views/Editor.vue:568]: This caused issue when user navigated to Datasets view and returned back to the same dataset, the UI was not in sync anymore.

Comment (tahme) [reply src/views/Editor.vue:568]: Removing this causes unsaved changes to be lost when you go to the Datasets view and return back.

Comment (jppiiroinen) [reply src/views/Editor.vue:568]: I think that we can live with that, what we should do is to add a dialog which warns the user that all unsaved changes are lost

File: src/views/Editor.vue [hunk 4]
```diff
@@ -310,14 +336,15 @@ export default {
 					await apiClient.put("/datasets/" + currentId, payload)
 					const { data } = await apiClient.get(`/datasets/${currentId}`)
 					this.qvainData = data
+					this.clearRecord()
+					this.openRecord(this.id)
```
Comment (tahme) [inline src/views/Editor.vue]: I like to save often and the editor reload and loss of UI state (scroll position, open tabs) this causes is somewhat disorienting. What does adding these fix?

Comment (jppiiroinen) [reply src/views/Editor.vue]: hmm

Comment (jppiiroinen) [reply src/views/Editor.vue]: i think that this has not been done on purpose

Comment (jppiiroinen) [reply src/views/Editor.vue]: this has been reverted

File: src/views/Editor.vue (hunk 3 above)
Comment (jppiiroinen) [reply src/views/Editor.vue:568]: the dialog has been added

File: src/views/Editor.vue (hunk 1 above)
Comment (tahme) [reply src/views/Editor.vue:159]: a small top margin could be nice so it wouldn't be touching the header

**Review by tahme — APPROVED** [review]

File: src/views/Editor.vue (hunk 1 above)
Comment (jppiiroinen) [reply src/views/Editor.vue:159]: I will add that one

**Pull request merged**